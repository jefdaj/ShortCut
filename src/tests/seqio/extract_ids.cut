fna = load_fna "data/6803genes10.fna"
faa = load_faa "data/6803genes10.faa"
result = any [extract_ids fna, extract_ids faa]

# TODO write this:
# mapped = extract_ids_each [fna]
# result = [single] | mapped
