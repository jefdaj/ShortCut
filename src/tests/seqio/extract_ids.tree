.
|-- cache
|   |-- each
|   |   |-- 10972a0db1
|   |   |   |-- 237bf736b9.str.list -> $TMPDIR/exprs/extract_ids/237bf736b9.str.list
|   |   |   `-- 237bf736b9.str.list.args
|   |   `-- e2d1834580
|   |       |-- 89c1f12bb3.str.list -> $TMPDIR/exprs/extract_ids/89c1f12bb3.str.list
|   |       `-- 89c1f12bb3.str.list.args
|   `-- list
|       |-- 0815de9417.txt
|       |-- 3c431f121f.txt
|       |-- 3d74a0df3f.txt
|       |-- 54a2bf7098.txt
|       |-- 88aa4b2fae.txt
|       |-- 8d8f73bf83.txt
|       |-- c8be8722f2.txt
|       |-- eb68d94c66.txt
|       `-- f2b04c697f.txt
|-- exprs
|   |-- any
|   |   |-- b9c67f5ccd.str.list.list
|   |   |-- cd540a98f9.str.list.list
|   |   `-- e981bf440f.str.list.list
|   |-- extract_ids
|   |   |-- 237bf736b9.str.list
|   |   |-- 6e8bd8b7c6.str.list
|   |   |-- 89c1f12bb3.str.list
|   |   `-- 8df2fbb1a8.str.list
|   |-- extract_ids_each
|   |   |-- 320f081082.str.list.list
|   |   `-- 34a61e93f2.str.list.list
|   |-- list
|   |   |-- 36c6fe3446.str.list.list -> ../../cache/list/54a2bf7098.txt
|   |   |-- 66fce79ea8.faa.list -> ../../cache/list/f2b04c697f.txt
|   |   |-- 70e8c48e15.fna.list -> ../../cache/list/3d74a0df3f.txt
|   |   |-- 8bbafcd642.str.list.list.list -> ../../cache/list/eb68d94c66.txt
|   |   |-- 9cc3a85fcd.str.list.list.list -> ../../cache/list/88aa4b2fae.txt
|   |   |-- b7cc684382.str.list.list -> ../../cache/list/8d8f73bf83.txt
|   |   `-- e34fcef70b.str.list.list.list -> ../../cache/list/3c431f121f.txt
|   |-- load_faa
|   |   `-- bb54df8bd7.faa -> $WORKDIR/data/6803genes10.faa
|   |-- load_fna
|   |   `-- 9b7934c579.fna -> $WORKDIR/data/6803genes10.fna
|   `-- str
|       |-- 50dbefb6d6.str -> ../../cache/list/c8be8722f2.txt
|       `-- 7da80bde02.str -> ../../cache/list/0815de9417.txt
`-- vars
    |-- bothFaa.str.list.list -> ../exprs/any/b9c67f5ccd.str.list.list
    |-- bothFna.str.list.list -> ../exprs/any/e981bf440f.str.list.list
    |-- faa.faa -> ../exprs/load_faa/bb54df8bd7.faa
    |-- fna.fna -> ../exprs/load_fna/9b7934c579.fna
    |-- mappedFaa.str.list.list -> ../exprs/extract_ids_each/320f081082.str.list.list
    |-- mappedFna.str.list.list -> ../exprs/extract_ids_each/34a61e93f2.str.list.list
    |-- result -> ../exprs/any/cd540a98f9.str.list.list
    |-- singleFaa.str.list -> ../exprs/extract_ids/6e8bd8b7c6.str.list
    `-- singleFna.str.list -> ../exprs/extract_ids/8df2fbb1a8.str.list

14 directories, 42 files
