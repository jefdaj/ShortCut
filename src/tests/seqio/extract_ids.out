[["SYNGTS_0001",
  "SYNGTS_0002",
  "SYNGTS_0003",
  "SYNGTS_0004",
  "SYNGTS_0005",
  "SYNGTS_0006",
  "SYNGTS_0007",
  "SYNGTS_0008",
  "SYNGTS_0009",
  "SYNGTS_0010"],
 ["AOY38_00005",
  "AOY38_00010",
  "AOY38_00015",
  "AOY38_00020",
  "AOY38_00025",
  "AOY38_00030",
  "AOY38_00035",
  "AOY38_00040",
  "AOY38_00045",
  "AOY38_00050"],
 ["AOY38_00005",
  "AOY38_00010",
  "AOY38_00015",
  "AOY38_00020",
  "AOY38_00025",
  "AOY38_00030",
  "AOY38_00035",
  "AOY38_00040",
  "AOY38_00045",
  "AOY38_00050"],
 ["SYNGTS_0001",
  "SYNGTS_0002",
  "SYNGTS_0003",
  "SYNGTS_0004",
  "SYNGTS_0005",
  "SYNGTS_0006",
  "SYNGTS_0007",
  "SYNGTS_0008",
  "SYNGTS_0009",
  "SYNGTS_0010"]]
